### PR TITLE
Switch to codecov.io

### PR DESCRIPTION
**Pull request by daspecster**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ script:
   - scripts/update_docs.sh
 
 after_success:
-  - tox -e coveralls
+  - bash <(curl -s https://codecov.io/bash)
```
Comment (dhermes) [inline .travis.yml]: Make a tox rule for this

Comment (dhermes) [inline .travis.yml]: Unless they have a "recommended best practice"

You should also remove the `coveralls` rule in `tox.ini`

Comment (daspecster) [inline .travis.yml]: That's how they say to do it.

screen shot 2016-06-20 at 4 31 36 pm

Comment (daspecster) [inline .travis.yml]: That's how they say to do it.
screen shot 2016-06-20 at 4 32 31 pm

File: tox.ini [hunk 2]
```diff
@@ -61,18 +61,10 @@ deps =
...
-    {[testenv]covercmd}
-    coveralls
-deps =
-    {[testenv:cover]deps}
-    coveralls
-setenv =
-    PYTHONPATH =
-passenv = {[testenv:system-tests]passenv}
+[testenv:codecov]
+passenv = CI TRAVIS_BUILD_ID TRAVIS TRAVIS_BRANCH TRAVIS_JOB_NUMBER TRAVIS_PULL_REQUEST TRAVIS_JOB_ID TRAVIS_REPO_SLUG TRAVIS_COMMIT
```
Comment (dhermes) [inline tox.ini]: Suggest going multiline here or just using a wildcard: `CI TRAVIS_*`

File: tox.ini [hunk 3]
```diff
@@ -61,18 +61,10 @@ deps =
...
+deps = codecov>=1.4.0
+commands = codecov -e TOXENV
```
Comment (dhermes) [inline tox.ini]: `TOXENV` here is just a string. Link?

Comment (daspecster) [inline tox.ini]: https://github.com/codecov/example-python

Comment (dhermes) [inline tox.ini]: That is just an example. From the CLI tool help:

```
  --env [ENV [ENV ...]], -e [ENV [ENV ...]]
                        Store environment variables to help distinguish CI
                        builds. Example: http://bit.ly/1ElohCu
```

This means `-e` is optional. We only run coverage in Python 2.7 so we don't need to distinguish.

**Comment by daspecster** [comment]
Possibly an example when it finishes generating...
https://codecov.io/github/daspecster/gcloud-python/commit/8f90abc07f33c0b84522ae58377cdcbaca569f65

**Comment by daspecster** [comment]
Yeah still playing with it.
I need to get the coverage xml generated before codecov runs.

On Monday, June 20, 2016, Codecov <notifications@github.com> wrote:

> Current coverage
> <https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/pull/1876?src=pr>
> is *100%*
>
> No coverage report found for *master* at 986302f
> <https://github.com/GoogleCloudPlatform/gcloud-python/commit/986302f3db5ab71e0951fa4fab796333d7d7cece>
> .
>
> Powered by Codecov <https://codecov.io?src=pr>. Last updated by
> 986302f...60c0364
> <https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/compare/986302f3db5ab71e0951fa4fab796333d7d7cece...60c03644ff146537e351822b77bd4a27732c06aa>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/1876#issuecomment-227272950>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALvyGawrhgLR23ki28X9RDdsOmdeqt7ks5qNwP7gaJpZM4I6FhG>
> .
>


-- 
Tom Schultz

**Comment by daspecster** [comment]
@dhermes the last build worked.

Travis:
https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/139037369#L778

codecov:
https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/tree/60c03644ff146537e351822b77bd4a27732c06aa/gcloud

File: tox.ini [hunk 4]
```diff
@@ -61,18 +61,21 @@ deps =
...
-basepython = {[testenv:cover]basepython}
-commands =
-    pip install gcloud[grpc]
-    {[testenv]covercmd}
-    coveralls
+[testenv:codecov]
+basepython =
+    python2.7
+passenv =
+    CI TRAVIS_BUILD_ID TRAVIS TRAVIS_BRANCH TRAVIS_JOB_NUMBER TRAVIS_PULL_REQUEST TRAVIS_JOB_ID TRAVIS_REPO_SLUG TRAVIS_COMMIT
```
Comment (dhermes) [inline tox.ini]: Past comment still applied. Either use multiline or `CI TRAVIS_*`

Comment (daspecster) [inline tox.ini]: Oh sorry, missed that.

File: tox.ini [hunk 5]
```diff
@@ -61,18 +61,21 @@ deps =
...
 deps =
-    {[testenv:cover]deps}
-    coveralls
+    {[testenv]deps}
+    {[grpc]deps}
+    coverage
+    codecov>=1.4.0
```
Comment (dhermes) [inline tox.ini]: It's clearer to just use

```
deps =
    {[testenv:cover]deps}
    codecov>=1.4.0
```

File: tox.ini [hunk 6]
```diff
@@ -61,18 +61,21 @@ deps =
...
 
-[testenv:coveralls]
-basepython = {[testenv:cover]basepython}
-commands =
-    pip install gcloud[grpc]
-    {[testenv]covercmd}
-    coveralls
+[testenv:codecov]
+basepython =
+    python2.7
```
Comment (dhermes) [inline tox.ini]: Ditto here on being clear to be the same env as cover: `basepython = {[testenv:cover]basepython}`

I'm just copy-pasting what was in `coveralls` for this one

**Comment by dhermes** [comment]
LGTM, please squash. Also we should make sure codecov.io adds a check to this PR before merging

**Comment by dhermes** [comment]
Also the coveralls badge should be replaced

**Comment by daspecster** [comment]
@dhermes, I don't have admin rights to the repo to add a github check.

**Comment by daspecster** [comment]
Looks like we're not alone!
https://codecov.io/gh/googlecloudplatform

**Comment by dhermes** [comment]
@daspecter Which page do you need admin rights for? I'm on https://github.com/GoogleCloudPlatform/gcloud-python/settings/hooks trying to add the service but can't find it

**Comment by dhermes** [comment]
OK figured it out on https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/settings

Here is the badge FYI

```

.. image:: https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/branch/master/graph/badge.svg
  :target: https://codecov.io/gh/GoogleCloudPlatform/gcloud-python
```

**Comment by dhermes** [comment]
@daspecster Now that the hook has been added, can you add a commit with the badge so we can test it?

**Comment by daspecster** [comment]
Awesome! I was looking.

I put `http://codecov.io/github/googlecloudplatform/gcloud-python/coverage.svg?branch=master` for the badge target in the readme but I'll change it to yours.

**Comment by dhermes** [comment]
I just copied that RST snippet from https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/settings/badge Where'd you get the other one?

**Comment by daspecster** [comment]
I think I saw it in their docs somewhere.

**Comment by dhermes** [comment]
```
codecov runtests: commands[1] | codecov
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.5
==> Detecting CI provider
  -> Got branch from git/hg
  -> Got sha from git/hg
==> Preparing upload
Error: Missing repository upload token
Tip: See all example repositories: https://github.com/codecov?query=example
Support channels:
  Email:   hello@codecov.io
  IRC:     #codecov
  Gitter:  https://gitter.im/codecov/support
  Twitter: @codecov
```

**Comment by dhermes** [comment]
Still no check, though [it worked][1]

```
codecov runtests: commands[1] | codecov
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.5
==> Detecting CI provider
    Travis Detected
==> Preparing upload
==> Processing gcov (disable by -X gcov)
    Executing gcov (find /home/travis/build/GoogleCloudPlatform/gcloud-python -type f -name '*.gcno'  -exec gcov -pb  {} +)
==> Collecting reports
    Generating coverage xml reports for Python
    + /home/travis/build/GoogleCloudPlatform/gcloud-python/coverage.xml bytes=1274444
==> Appending environment variables
    + TRAVIS_OS_NAME
==> Uploading
    .url https://codecov.io
    .query commit=51af5e55de35c61123d61d8cc1eb76eca06b3333&service=travis&pr=1876&package=py2.0.5&job=139057017&branch=master&slug=GoogleCloudPlatform%2Fgcloud-python&build=3845.1
    Pinging Codecov...
    Uploading to S3...
```

[1]: https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/139057015

**Comment by daspecster** [comment]
Hmmm

**Comment by daspecster** [comment]
@dhermes after some digging, I think we need that codecov.yml file there to get it to trigger status checks. Let me know if one shows up now?

**Comment by dhermes** [comment]
Still only 2 checks

**Comment by daspecster** [comment]
Ok, I'll see what else I can find. Thanks for checking.

**Comment by daspecster** [comment]
It looks like it's getting a 404 for something but I can't tell if this is a codecov.io URL or a github url that's throwing the error.

https://codecov.io/gh/googlecloudplatform/gcloud-python/commit/db465386c9819fc384c2c6f62cb2ccdd79566be7/builds

**Comment by daspecster** [comment]
Update: I have a support request out to codecov.io. I think it might be a permissions issue but I want to confirm with them so I know how it works.

**Comment by daspecster** [comment]
@dhermes, can you add the integration here https://github.com/integrations/codecov to the GoogleCloudApplication org if possible.
There might be something to do with the third-party restrictions as well, possibly in the org though.

_I think_ https://github.com/organizations/GoogleCloudPlatform/settings/oauth_application_policy
and
https://github.com/organizations/GoogleCloudPlatform/settings/applications/

**Comment by dhermes** [comment]
I don't know if I have admin rights / don't see any "Add" button on the integrations page (I visited it multiple times yesterday trying to figure out what to click).

**Comment by daspecster** [comment]
Would this be something @jgeewax might have to do?

**Comment by dhermes** [comment]
Maybe

**Comment by dhermes** [comment]
@daspecster Where did the previous [post][1] from the codecov bot come from?

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/pull/1876#issuecomment-227272950

**Comment by daspecster** [comment]
If you change this value it will post a comment.
https://github.com/GoogleCloudPlatform/gcloud-python/pull/1876/files#diff-8bc9db6adba7208e25c96be322351063R1

**Comment by dhermes** [comment]
OK

**Comment by daspecster** [comment]
@jgeewax, if you're game, can we enable codecov in the organization apps/oauth?

**Comment by jgeewax** [comment]
👍

**Comment by daspecster** [comment]
Hmmm, still not showing up in here. Could someone check and see if codecov is in the projected branch settings now?

**Comment by daspecster** [comment]
Looks like there is still an issue on the github permissions side.
screen shot 2016-07-13 at 2 44 31 pm

**Comment by tseaver** [comment]
FWIW, the webhook is already in place.

screenshot from 2016-07-14 10 54 57

**Comment by daspecster** [comment]
@tseaver, if you go [here](https://codecov.io/gh/GoogleCloudPlatform/gcloud-python/settings) can you click on the "reset webhook" button with any success?

If that doesn't start working then we could also just turn the integration back on to get codecov to post a comment about the coverage.

I'm thinking that this might have to get into [master](https://codecov.io/gh/Googlecloudplatform/gcloud-python/branches) to fully work. We could change this so it does both coveralls and codecov to test it.

**Comment by tseaver** [comment]
> Can you click on the "reset webhook" button with any success?

Nope, I get the same error you posted.

> If that doesn't start working then we could also just turn the integration back on to get codecov to post a comment about the coverage.

I don't like those comments:  I would rather have it be a "check".

> I'm thinking that this might have to get into master to fully work. We could change this so it does both coveralls and codecov to test it.

Let's try that.  We can drop coveralls in a follow-up PR.

**Comment by tseaver** [comment]
LGTM

**Comment by daspecster** [comment]
I squashed some of the commits and when travis passes I'll merge.

**Pull request merged**